src/rpc/blockchain.cpp

```diff
@@ -58,10 +58,7 @@ static CUpdatedBlock latestblock;
  */
 double GetDifficulty(const CBlockIndex* blockindex)
 {
-    if (blockindex == nullptr)
-    {
-        return 1.0;
-    }
+    assert(blockindex);
```

Comment:
In rpc code, assert should be replaced with throw JSONRPCError?

Reply:
I don't think this is a usage error, should be a programatic error?

Reply:
I guess you could avoid the assert by passing in a const reference?

Reply:
Sure but out of scope here and I'm happy to submit that refactor in a separate PR.